src/components/TodoApp/TodoApp.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useContext, useMemo, useState } from 'react';
+import { TodoList } from '../TodoList';
+import { Filter } from '../Filter';
+import { Filters } from '../../types/enumFilter';
+import { prepareTodos } from '../../utils/prepareTodos';
+import { TodosContext } from '../../context/TodosContext';
+import { AddTodoForm } from '../AddTodoForm';
+
+export const TodoApp: React.FC = () => {
+  const [filterField, setFilterField] = useState(Filters.All);
+
+  const {
+    todos,
+    deleteCompletedTodo,
+    updateTodo,
+  } = useContext(TodosContext);
+
+  const someCompleted = todos.some(todo => todo.completed);
+  const everyCompleted = todos.every(todo => todo.completed);
+
+  const preparedTodos = useMemo(
+    () => prepareTodos(todos, filterField),
+    [todos, filterField],
+  );
+
+  const countNotCompletedTodo = React.useMemo(() => {
```

Comment:
Note - that useMemo is redundant here
```suggestion
  const countNotCompletedTodo = useMemo(() => {
```